src/Common/SurfaceFluxes/SurfaceFluxes.jl

```diff
@@ -37,366 +31,296 @@
+    println(io, "K_exchange        = ", sfc.K_exchange)
+    println(io, "-----------------------")
 end
 
 """
-    compute_friction_velocity(param_set, flux, z_0, z_1, β_m, γ_m, tol_abs, iter_max)
-
-Computes roots of friction velocity equation (Eq. 10)
-
-`u_ave = u_* ( ln(z/z_0) - ψ_m(z/L, z_0/L) ) /κ`
+    surface_conditions
+Surface conditions given
+ - `x_initial` initial guess for solution (`L_MO, u_star, θ_star, ϕ_star, ...`)
+ - `x_ave` volume-averaged value for variable `x`
+ - `x_s` surface value for variable `x`
+ - `z_0` roughness length for variable `x`
+ - `F_exchange` flux at the top for variable `x`
+ - `dimensionless_number` dimensionless turbulent transport coefficient:
+      - Momentum: 1
+      - Heat: Turbulent Prantl number at neutral stratification
+      - Mass: Turbulent Schmidt number
+      - ...
+ - `θ_bar` basic potential temperature
+ - `Δz` layer thickness (not spatial discretization)
+ - `z` coordinate axis
+ - `a` free model parameter with prescribed value of 4.7
+ - `pottemp_flux_given` potential temperature flux (optional)
+
+If `pottemp_flux` is not given, then it is computed by iteration
+of equations 3, 17, and 18 in Nishizawa2018.
 """
-function compute_friction_velocity(
-    param_set::AbstractParameterSet,
-    u_ave::FT,
-    flux::FT,
-    z_0::FT,
-    z_1::FT,
-    β_m::FT,
-    γ_m::FT,
-    tol_abs::FT,
-    iter_max::IT,
-) where {FT, IT}
-
-    _von_karman_const = FT(von_karman_const(param_set))
-
-    ustar_0 = u_ave * _von_karman_const / log(z_1 / z_0)
-    ustar = ustar_0
-    let u_ave = u_ave,
-        flux = flux,
-        z_0 = z_0,
-        z_1 = z_1,
-        β_m = β_m,
-        γ_m = γ_m,
-        param_set = param_set
-
-        # use neutral condition as first guess
-        stable = z_1 / monin_obukhov_len(param_set, ustar_0, flux) >= 0
-        function compute_ψ_m(u)
-            L_MO = monin_obukhov_len(param_set, u, flux)
-            ζ = z_1 / L_MO
-            ζ_0 = z_0 / L_MO
-            return stable ? ψ_m_stable(ζ, ζ_0, β_m) : ψ_m_unstable(ζ, ζ_0, γ_m)
-        end
-        function compute_u_ave_over_ustar(u)
-            _von_karman_const = FT(von_karman_const(param_set))
-            return (log(z_1 / z_0) - compute_ψ_m(u)) / _von_karman_const # Eq. 10
-        end
-        compute_ustar(u) = u_ave / compute_u_ave_over_ustar(u)
-
-        if (abs(flux) > 0)
-            ustar_1 = compute_ustar(ustar_0)
-            sol = RootSolvers.find_zero(
-                u -> u_ave - u * compute_u_ave_over_ustar(u),
-                SecantMethod(ustar_0, ustar_1),
-                CompactSolution(),
-                SolutionTolerance(tol_abs),
-                iter_max,
-            )
-            ustar = sol.root
+function surface_conditions(
+    param_set::AbstractEarthParameterSet,
+    x_initial::Vector{FT},
+    x_ave::Vector{FT},
+    x_s::Vector{FT},
+    z_0::Vector{FT},
+    F_exchange::Vector{FT},
+    dimensionless_number::Vector{FT},
+    θ_bar::FT,
```

Comment:
Same as above: Replace potential temperature by dry static energy (with DSE containing the water vapor buoyancy term).